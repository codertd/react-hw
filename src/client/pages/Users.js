import React from 'react';
import PropTypes from 'prop-types';


const Users = (props) => {
  const { text } = props;
  return (
    <users>
      <h1 data-testid="h1tag" className="fancy-h1">
        {text}
      </h1>
    </users>
  );
};

Users.propTypes = {
  text: PropTypes.string.isRequired,
};

export default Users;
