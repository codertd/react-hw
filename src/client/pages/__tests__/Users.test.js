// __tests__/Users.test.js

import React from 'react';
import {
  render
} from '@testing-library/react';
import Users from '../Users';

it('renders', () => {
  const { asFragment, getByTestId, getByText } = render(<Users text="Hello!" />);
  expect(asFragment()).toMatchSnapshot();

  expect(getByTestId('h1tag')).toHaveTextContent('Hello!');
  expect(getByText('Hello!')).toHaveClass('fancy-h1');
});

// test('Component Renders', () => {
//   render(<Users>Users</Users>);

//   // query* functions will return the element or null if it cannot be found
//   // get* functions will return the element or throw an error if it cannot be found
//   // expect(screen.queryByText('Users')).toBe(<h1>Users</h1>);

//   // the queries can accept a regex to make your selectors more resilient to content tweaks
//   // and changes.
//   // fireEvent.click(screen.getByLabelText(/show/i));

//   // .toBeInTheDocument() is an assertion that comes from jest-dom
//   // otherwise you could use .toBeDefined()
//   expect(screen.getByText('Users')).toBeInTheDocument();
// });
