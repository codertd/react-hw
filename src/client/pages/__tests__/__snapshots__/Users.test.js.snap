// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <users>
    <h1
      class="fancy-h1"
      data-testid="h1tag"
    >
      Hello!
    </h1>
  </users>
</DocumentFragment>
`;
